### PR TITLE
Rename source acceptance test to connector acceptance test

### LINKED ISSUE #21382: Rename SAT to Connector acceptance tests
After https://github.com/airbytehq/airbyte/issues/20063 will be done, the spec part of the SATs are a very useful tool beyond only source connectors as they do basic checks about whether the spec is confirming to all documented requirements and will not cause problems in the UI as well.

The SATs are operating on the docker interface which means the spec test can be run without any changes against destination connector images. However, the name "source acceptance tests" is confusing in the context of destination connectors. As there are already multiple test suites for connectors which can be confusing for novices (e.g. standard destination tests), the source acceptance tests should be renamed to connector acceptance test to better reflect their more general nature. This should be done before spec acceptance tests are rolled out for destination connectors.

## Acceptance criteria

* Source acceptance tests (SAT) are renamed to connector acceptance tests in all places
  * Directory/file names
  * Python classes
  * Documentation
  * Groovy plugin
* Documentation page is adjusted to document new structure
* https://docs.airbyte.com/connector-development/testing-connectors/source-acceptance-tests-reference/ links to new `connector-acceptance-tests-reference` page

### PR DESCRIPTION
**Pull request by flash1293**
## What

Fixes https://github.com/airbytehq/airbyte/issues/21382

This PR renames the SATs to simply (connector) acceptance tests. The "connector" prefix can be dropped in descriptions if there is no ambiguity. This is done to make it clear that parts of the provided test suites (the spec-related tests) are general enough to also be used for destinations. In a separate step these will be rolled out for existing destinations.

The general way of configuring tests via the `acceptance-test-config.yml` doesn't change.

_Note: Originally I wanted to remove the `connector-` prefix in more places, but there are places where I would strongly prefer it (e.g. in the name of the docker image to make clear what it is about or the gradle plugin as there are also kubernetes acceptance tests at the moment) and it seemed weird to me to switch the naming over in some place so I stuck with it in most places for now (e.g. name of the python package)._

## How
* Source acceptance tests (SAT) are renamed to connector acceptance tests in all places (directory names of the code, docker image, gradle plugin and tasks)
* The new docker image `airebyte/connector-acceptance-tests` will start with the same version as the existing one to ease migration. Existing references to the old image will still work, they just won't receive upgrades anymore. (it's not published yet, I will do this once this PR is approved)
* Documentation page is adjusted to document new structure
  * Existing pages referencing source acceptance tests directly are renamed
  * Test this by running `yarn && yarn build && yarn serve` in the `docusaurus` directory

## How to review

* `airbyte-integrations/bases` - rename `source-acceptance-test` directory
* `airbyte-integrations/connector-templates` - reference new docker image / gradle task
* `airbyte-integrations/connectors`
  * Switch to new docker image in helper shell script and adjust comment in yml file
  * Switch to new gradle task* `docs`
  * Existing pages referencing source acceptance tests directly are renamed
  * There is a redirect page for the existing source-acceptance-test page to teach people it got renamed
  * Test this by running `yarn && yarn build && yarn serve` in the `docusaurus` directory
* `buildSrc` rename gradle plugin and make it use the new image
* `settings.gradle` change name of task for building acceptance tests themselves
* `tools` rename references

### CODE REVIEW

**Comment by flash1293** [comment]
edited by github-actions bot 
/test connector=connectors/source-stripe

**Review by evantahler — COMMENTED** [review]
1. I like this initiative! 
2. I don't have any idea how to test this PR without running `/test` on all the effected connectors... and for some, the tests are expected to fail for other reasons.  This might need to be one of those YOLO PRs if we gain enough confidence with the GA connectors
3. We should `/test` all GA connectors maybe?  Or, since this is not something that will take down production, we could YOLO it once again and see what the nightly tests in #connector-health report...

File: docs/connector-development/testing-connectors/source-acceptance-tests-reference.md [hunk 1]
```diff
@@ -1,360 +1,5 @@
...
-          missing_streams:
-            - name: my_missing_stream
-              bypass_reason: "Please fill a good reason"
-```
-
-## Caching
-
-We cache discovered catalogs by default for performance and reuse the same discovered catalog through all tests.
-You can disable this behavior by setting `cached_discovered_catalog: False` at the root of the configuration.
+You can find the recent documentation [here](./connector-acceptance-tests-reference.md)
```
Comment (girarda) [inline docs/connector-development/testing-connectors/source-acceptance-tests-reference.md:5]: keeping this file and linking to the new one is a good idea!

File: tools/ci_connector_ops/ci_connector_ops/sat_config_checks.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 REVIEW_REQUIREMENTS_FILE_PATH = ".github/connector_org_review_requirements.yaml"
 
 def find_connectors_with_bad_strictness_level() -> List[utils.Connector]:
-    """Check if changed connectors have the expected SAT test strictness level according to their release stage.
+    """Check if changed connectors have the expected acceptance test strictness level according to their release stage.
```
Comment (girarda) [inline tools/ci_connector_ops/ci_connector_ops/sat_config_checks.py:19]:
```suggestion
    """Check if changed connectors have the expected connector acceptance test strictness level according to their release stage.
```

**Review by girarda — APPROVED** [review]
left a few suggestions, but nothing blocking on my end.

Approved pending testing. I agree with Evan that running /test on the GA connectors should be enough since failures won't affect production

File: .github/workflows/sat_strictness_level.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: Connector Ops CI - Check SAT strictness level
+name: Connector Ops CI - Check acceptance test strictness level
```
[the remark this answers is not in the document]
Comment (flash1293) [reply .github/workflows/sat_strictness_level.yml:1]: Got renamed in the meantime to Connector Metadata Checks

File: airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py [hunk 4]
```diff
@@ -348,7 +348,7 @@ def pytest_sessionfinish(session, exitstatus):
         print()  # create a line break
         logger.info(
             # session.startdir gives local path to the connector folder, so we can verify which cnctr was tested
-            f"{session.startdir} - SAT run - "
+            f"{session.startdir} - CAT run - "
```
[the remark this answers is not in the document]
Comment (flash1293) [reply airbyte-integrations/bases/connector-acceptance-test/connector_acceptance_test/conftest.py:351]: I don't think we should start with "CAT", this is a leftover. Went with connector acceptance tests here

**Comment by flash1293** [comment]
The runs above failed, but I think this is caused by https://github.com/airbytehq/airbyte/issues/22127 - I pulled in https://github.com/airbytehq/airbyte/pull/22131 which seems to make the builds work for now. FYI @bechurch so you are aware of this work

**Comment by flash1293** [comment]
@evantahler @bechurch I tested on a clean master state and I'm getting exactly the same error as on this PR when trying to run connectors - the tests run fine but the QA checks fail: https://github.com/airbytehq/airbyte/pull/22136#issuecomment-1410148433

Could you take a look at that?

**Comment by flash1293** [comment]
@evantahler It seems like the bot is currently a little hesitant to schedule my tests - I will try again tomorrow morning, but feel free to look from a code side. It seems like this works pretty fine and I would like to get it in quickly as there's a lot of surface for merge conflicts.

**Review by grishick — APPROVED** [review]
LGMT pending passing CI checks

**Comment by evantahler** [comment]
@flash1293 I think we are having rate limit issues... It looks like your `/test`s are finally getting 👀...

**Review by evantahler — APPROVED** [review]
I've got low-confidence that my review was thorough on all these files... but these changes are also unlikely to harm production if we got something wrong. I think the added clarity is worth the risk! Thanks @flash1293!

**Comment by flash1293** [comment]
/publish connector=bases/connector-acceptance-test auto-bump-version=false

**Pull request merged**